Documentation/ExtensionArchitecture/Extbase/Reference/_FileUpload/_Blog.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Domain\Model;
+
+use TYPO3\CMS\Extbase\Domain\Model\FileReference;
+use TYPO3\CMS\Extbase\DomainObject\AbstractEntity;
+use TYPO3\CMS\Extbase\Persistence\ObjectStorage;
+
+class Blog extends AbstractEntity
+{
+    // A single file
+    protected ?FileReference $singleFile = null;
+
+    /**
+     * A collection of files.
+     * @var ObjectStorage<FileReference>
+     */
+    protected ObjectStorage $multipleFiles;
+
+    // When using ObjectStorages, it is vital to initialize these.
+    public function __construct()
+    {
+        $this->initializeObjectStorage();
+    }
+
+    public function initializeObjectStorage(): void
+    {
+        $this->multipleFiles = new ObjectStorage();
+    }
```

Comment:
```suggestion
    // When using ObjectStorages, it is vital to initialize these.
    public function __construct()
    {
        $this->multipleFiles = new ObjectStorage();
    }

    /**
     * Called again with initialize object, as fetching an entity from the DB does not use the constructor
     */
    public function initializeObject(): void
    {
        $this->multipleFiles = $this->multipleFiles ?? new ObjectStorage();
    }
```

Reply:
I created this with the "new object" in mind, but I see your point. Haven't verified this, but I'll take your suggestion for now and cross my fingers ;-)